### ISSUE TITLE
PHP Include Handler Assumes Relative URLs

### ISSUE THREAD

**Issue by ssigwart** [Komodo/KomodoEdit#3710]
### Short Summary
The following code only works for relative include paths or absolute paths.
https://github.com/Komodo/KomodoEdit/blob/3a6fd631da21003af89b9f9bb855d936c753411c/src/chrome/komodo/content/hyperlinks/filehandler.js#L220-L223

### Steps to Reproduce
Set up a PHP project with `www` and `includes` top level folders.  Add `www/index.php` which includes `require_once('MyClass.class.php');` . Add `includes/MyClass.class.php`.  In `www/index.php`, click the hyperlink to `MyClass.class.php`.

### <strike>Expected</strike> Desired results
`includes/MyClass.class.php` should open.

### Actual results
Error message that file can't be found.

### Platform Information
Doesn't matter.

### Additional Information
Ideally, it would know the PHP include paths, but that's impractical since `set_include_path` can be used anywhere in the PHP code.  @cgchoffman or @Naatan, do you have any objections to me adding a Project level preference to add a list of include paths?  Then I can update `filehandler.js` to use `alternativePaths` of `filename_jump_handler`.

**Comment by cgchoffman** [comment]
@ssigwart, can you use Prefs > Languages > PHP > PHP Directories?

**Comment by ssigwart** [comment]
I think I'd run into problems there.  The directories are already included and scanned, it's just that `filename_jump_handler` doesn't know the exact location of the files.  I didn't test it, but I'm pretty sure it doesn't look at that setting to find the file.

**Comment by cgchoffman** [comment]
> but I'm pretty sure it doesn't look at that setting to find the file.

yeah exactly.  I was thinking change `filename_jump_handler` to use it but it seems so specific to have a pref that says "I'll have files here, and here, and here and here".  I wonder if this could hook into the Commando Files scope.  @Naatan, I just had a look at the Files scope code to see if it could be used here but the search code that takes a `onComplete` function doesn't receive anything from the search.  

Is there a way that code that's not part of that scope to use the cached structure to quickly find a file in the project?
